monkey/monkey_island/cc/resources/agent_binaries.py

```diff
@@ -15,7 +17,8 @@ class AgentBinaries(AbstractResource):
     def __init__(self, agent_binary_repository: IAgentBinaryRepository):
         self._agent_binary_repository = agent_binary_repository
 
-    # Used by monkey. can't secure.
+    @auth_token_required
+    @roles_accepted(AccountRole.AGENT.name)
```

Comment:
Good candidate for both ISLAND and AGENT roles. Is better to have both roles then to have none.